tests/api/test_nanopubs.py

```diff
@@ -31,71 +26,54 @@ def test_create(self):
         self.assertEquals(len(nanopub.provenance), 0)
 
     def test_read(self):
-        self.login(*self.create_user("user@example.com","password"))
-        response = self.client.post("/pub", data=self.turtle, content_type="text/turtle",follow_redirects=True)
-        self.assertEquals(response.status,'201 CREATED')
+        response = self.publish_nanopub(self.turtle, "text/turtle")
         self.assertTrue('Location' in response.headers)
-        
+
         nanopub_id = response.headers['Location'].split('/')[-1]
-        
-        content = self.client.get("/pub/"+nanopub_id,headers={'Accept':'application/json'}, follow_redirects=True)
+        content = self.client.get("/pub/"+nanopub_id, headers={'Accept':'application/json'}, follow_redirects=True)
         g = ConjunctiveGraph()
         self.assertEquals(content.mimetype, "application/json")
-        g.parse(data=str(content.data,'utf8'), format="json-ld")
-        
+        g.parse(data=str(content.data, 'utf8'), format="json-ld")
+
         self.assertEquals(len(g), 17)
         self.assertEquals(g.value(URIRef('http://example.com/janedoe'), RDF.type), URIRef('http://schema.org/Person'))
 
     def test_delete_admin(self):
-        self.login(*self.create_user("user@example.com","password"))
```

Comment:
A good pattern here is a sort of ghetto partial evaluation: have a public method that takes a bunch of parameters and then a private method that fills some of them in.
def login on the parent class, unchanged
def __login(self): return self.login(*self.create_user("user@example.com", "password"))